plugin-examples/src/test/java/com/hotels/styx/EarlyReturnExamplePluginTest.java

```diff
@@ -0,0 +1,34 @@
+package com.hotels.styx;
+
+import com.hotels.styx.api.*;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+/**
+ * This tests the behaviours added in the example plugin.
+ */
+
+public class EarlyReturnExamplePluginTest {
```

Comment:
We should also include a test that proves the behaviour when the `X-Respond` header is not present.